packages/babel-parser/src/parser/statement.js

```diff
@@ -2068,6 +2087,33 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
```

Comment:
Do you think it's worth having two plugins? One for `as` and the other for the attributes

Reply:
We should have a single plugin with an option, similarly to the pipeline operator.

Reply:
@nicolo-ribaudo can you help me understand which file to look at to understand this option to allow different syntax with the same plugin?

Reply:
You can search for `getPluginOption(` in `expression.js`.

Reply:
Understood 👍

Reply:
Sure, that works too. We just need to make sure that both can be enabled at the same time

Reply:
Why? I don't think that alternative versions of the same proposal should be mixed.

Reply:
We don't see them as different versions of the same proposal. Imagine `as "json"` being a shorthand for `with type:"json"`. We are going to present that idea to the committee, but eventually they will live together. Does that make sense to you?

Reply:
Oh ok :+1:

Then we can have two boolean option, `as` and `with`.